node-git and node-gyp errors in `yarn install`

node-git and node-gyp errors in `yarn install`

## Expected Behavior
Build should be successful after following the correct setup instructions as per the documentation. 
e.g. Node v12, YARN as pre-requisites.

## Current Behavior
`yarn install` does not exit with success:

```
[4/4] Building fresh packages...
error D:\backstage\node_modules\nodegit: Command failed.
Exit code: 1
Command: node lifecycleScripts/preinstall && node lifecycleScripts/install
Arguments:
Directory: D:\backstage\node_modules\nodegit
Output:
[nodegit] Running pre-install script
[nodegit] Running install script
node-pre-gyp
info it worked if it ends with ok
node-pre-gyp
info using node-pre-gyp@0.13.0
node-pre-gyp info using node@12.18.4 | win32 | x64
node-pre-gyp WARN
Using request for node-pre-gyp https download
node-pre-gyp
info check checked for "D:\backstage\node_modules\nodegit\build\Release\nodegit.node" (not found)
node-pre-gyp
http GET https://axonodegit.s3.amazonaws.com/nodegit/nodegit/nodegit-v0.27.0-node-v72-win32-x64.tar.gz
node-pre-gyp
WARN Pre-built binaries not installable for nodegit@0.27.0 and node@12.18.4 (node-v72 ABI, unknown) (falling back to source compile with node-gyp)
node-pre-gyp WARN Hit error connect ETIMEDOUT 10.109.23.186:443
gyp
info
it worked if it ends with ok
gyp info
using node-gyp@4.0.0
gyp info using node@12.18.4 | win32 | x64
gyp
info ok
gyp
info
it worked if it ends with ok
gyp info
using node-gyp@4.0.0
gyp info using node@12.18.4 | win32 | x64
gyp
ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (D:\backstage\node_modules\node-gyp\lib\configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (D:\backstage\node_modules\node-gyp\lib\configure.js:509:16)
gyp ERR! stack     at callback (D:\backstage\node_modules\graceful-fs\polyfills.js:295:20)
gyp
ERR! stack     at FSReqCallback.oncomplete (fs.js:168:21)
gyp ERR! System Windows_NT 10.0.16299
gyp
ERR! command "C:\\Program Files\\nodejs\\node.exe" "D:\\backstage\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--module=D:\\backstage\\node_modules\\nodegit\\build\\Release\\nodegit.node" "--module_name=nodegit" "--module_path=D:\\backstage\\node_modules\\nodegit\\build\\Release" "--napi_version=6" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v72"
gyp ERR! cwd D:\backstage\node_modules\nodegit
gyp ERR! node -v v12.18.4
gyp ERR! node-gyp -v v4.0.0
gyp ERR!
not ok
node-pre-gyp
ERR! build error
node-pre-gyp
ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe D:\backstage\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --module=D:\backstage\node_modules\nodegit\build\Release\nodegit.node --module_name=nodegit --module_path=D:\backstage\node_modules\nodegit\build\Release --napi_version=6 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (D:\backstage\node_modules\nodegit\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp
ERR! stack     at ChildProcess.emit (events.js:315:20)
node-pre-gyp
ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR!
stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
node-pre-gyp
ERR! System Windows_NT 10.0.16299
node-pre-gyp
ERR! command "C:\\Program Files\\nodejs\\node.exe" "D:\\backstage\\node_modules\\nodegit\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp
ERR! cwd D:\backstage\node_modules\nodegit
node-pre-gyp ERR!
node -v v12.18.4
node-pre-gyp ERR!
node-pre-gyp -v v0.13.0
node-pre-gyp ERR! not ok
Failed to execute 'C:\Program Files\nodejs\node.exe D:\backstage\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --module=D:\backstage\node_modules\nodegit\build\Release\nodegit.node --module_name=nodegit --module_path=D:\backstage\node_modules\nodegit\build\Release --napi_version=6 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
[nodegit] ERROR - Could not finish install
[nodegit] ERROR - finished with error code: 1

```


Potentially due to that, when backend is started using `yarn start`, it throws MODULE_NOT_FOUND error for node-gyp.

```
$ yarn start
yarn run v1.22.10
$ backstage-cli backend:dev
Build succeeded
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module '../build/Debug/nodegit.node'
Require stack:
- D:\backstage\node_modules\nodegit\dist\nodegit.js
- D:\backstage\node_modules\@backstage\plugin-scaffolder-backend\dist\index.cjs.js
- D:\backstage\packages\backend\dist\main.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (D:\backstage\node_modules\nodegit\dist\nodegit.js:19:12)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\backstage\\node_modules\\nodegit\\dist\\nodegit.js',
    'D:\\backstage\\node_modules\\@backstage\\plugin-scaffolder-backend\\dist\\index.cjs.js',
    'D:\\backstage\\packages\\backend\\dist\\main.js'
  ]
}
```

## Steps to Reproduce

npx @backstage/create-app
Error
=================
running `yarn install` also gives the same error

## Context

Fresh install using both approaches : cli and cloning _backstage_ github repo.

## Your Environment

- NodeJS Version (v12): v12.18.4
- npm v6.14.6
- Operating System and Version: Windows 10, 64-bit
- Browser Information:

--- a/docs/getting-started/create-an-app.md
+++ b/docs/getting-started/create-an-app.md
@@ -38,6 +38,34 @@ app-folder is the name that was provided when prompted.
 Inside that directory, it will generate all the files and folder structure
 needed for you to run your app.
 
+### Troubleshooting
+
+The create app command doesn't always work as expected, this is a collection of
+some of the commonly encountered issues and solutions.
+
+#### Couldn't find any versions for "file-saver"
+
+You may encounter the following error message:
+
+```text
+Couldn't find any versions for "file-saver" that matches "eligrey-FileSaver.js-1.3.8.tar.gz-art-external"
+```
+
+This is likely because you have a globally configured NPM proxy, which breaks
+the installation of the `material-table` dependency. This is a known issue and
+being worked on in `material-table`, but for now you can work around it using
+the following:
+
+```bash
+NPM_CONFIG_REGISTRY=https://registry.npmjs.org npx @backstage/create-app
+```
+
+#### Can't find Python executable "python"
+
+The install process may also fail if no Python installation is available. Python
+is commonly available in most systems already, but if it isn't you can head for
+example [here](https://www.python.org/downloads/) to install it.
+
 ### General folder structure
 
 Below is a simplified layout of the files and folders generated when creating an

--- a/docs/getting-started/index.md
+++ b/docs/getting-started/index.md
@@ -24,9 +24,11 @@ Requests towards this repo.
 Backstage provides the `@backstage/create-app` package to scaffold standalone
 instances of Backstage. You will need to have
 [NodeJS](https://nodejs.org/en/download/) Active LTS Release installed
-(currently v12), and [yarn](https://classic.yarnpkg.com/en/docs/install). You
-will also need to have [Docker](https://docs.docker.com/engine/install/)
-installed to use some features like Software Templates and TechDocs.
+(currently v12), [yarn](https://classic.yarnpkg.com/en/docs/install) and
+[Python](https://www.python.org/downloads/) (although you likely have it
+already). You will also need to have
+[Docker](https://docs.docker.com/engine/install/) installed to use some features
+like Software Templates and TechDocs.
 
 Using `npx` you can then run the following to create an app in a chosen
 subdirectory of your current working directory:

--- a/docs/tutorials/quickstart-app-auth.md
+++ b/docs/tutorials/quickstart-app-auth.md
@@ -11,10 +11,11 @@ title: Monorepo App Setup With Authentication
 > own environment. It starts with a skeleton install and verifying of the
 > monorepo's functionality. Next, GitHub authentication is added and tested.
 >
-> This document assumes you have NodeJS 12 active along with Yarn. Please note,
-> that at the time of this writing, the current version is 0.1.1-alpha.21. This
-> guide can still be used with future versions, just, verify as you go. If you
-> run into issues, you can compare your setup with mine here >
+> This document assumes you have NodeJS 12 active along with Yarn and Python.
+> Please note, that at the time of this writing, the current version is
+> 0.1.1-alpha.21. This guide can still be used with future versions, just,
+> verify as you go. If you run into issues, you can compare your setup with mine
+> here >
 > [simple-backstage-app](https://github.com/johnson-jesse/simple-backstage-app).
 
 # The Skeleton Application

--- a/docs/tutorials/quickstart-app-plugin.md
+++ b/docs/tutorials/quickstart-app-plugin.md
@@ -20,10 +20,11 @@ title: Adding Custom Plugin to Existing Monorepo App
 > functionality, extend the Sidebar to make our life easy. Finally, we add
 > custom code to display GitHub repository information.
 >
-> This document assumes you have NodeJS 12 active along with Yarn. Please note,
-> that at the time of this writing, the current version is 0.1.1-alpha.21. This
-> guide can still be used with future versions, just, verify as you go. If you
-> run into issues, you can compare your setup with mine here >
+> This document assumes you have NodeJS 12 active along with Yarn and Python.
+> Please note, that at the time of this writing, the current version is
+> 0.1.1-alpha.21. This guide can still be used with future versions, just,
+> verify as you go. If you run into issues, you can compare your setup with mine
+> here >
 > [simple-backstage-app-plugin](https://github.com/johnson-jesse/simple-backstage-app-plugin).
 
 # The Skeleton Plugin

